option/syscall.c

```diff
@@ -126,14 +124,13 @@ void ff_rel_grant (
 /*------------------------------------------------------------------------*/
 /* Allocate a memory block                                                */
 /*------------------------------------------------------------------------*/
-/* If a NULL is returned, the file function fails with FR_NOT_ENOUGH_CORE.
-*/
+/* If a NULL is returned, the file function fails with FR_NOT_ENOUGH_CORE. */
 
 void* ff_memalloc (	/* Returns pointer to the allocated memory block */
 	UINT msize		/* Number of bytes to allocate */
 )
 {
-	return malloc(msize);	/* Allocate a new memory block with POSIX API */
+	return k_malloc(msize);	/* Allocate a new memory block with POSIX API */

```

Comment:
Is this acceptable, what if this code is running from user space?

I would rather keep malloc and put a dependency towards minimal libc with malloc or newlib.

Reply:
I can revert to using malloc/free if you think that is the safe option. Either way there is the compile-time option of using BSS or stack allocation for the FAT name buffers, which would not put such a dependency towards libc or newlib.

Reply:
@andrewboie Do you have any opinion on this?

Reply:
> Why did you make this change?
Misunderstanding on my part then. I made this change when looking at all #if's related to _USE_LFN and thought k_malloc was appropriate for subsys/fs.
Reverting to use malloc/free again.